### ISSUE TITLE
play.editor setting in application.conf is ignored

### ISSUE THREAD

**Issue by nemoo** [playframework/playframework#8330]
Scala Play Version 2.6.12
Windows 7
JDK Oracle 1.8.0_72

### Expected Behavior
According to [play framework documentation](https://www.playframework.com/documentation/2.6.x/IDE#Navigate-from-an-error-page-to-the-source-code):

"Using the play.editor configuration option, you can set up Play to add hyperlinks to an error page." 

The expectation is that in case of a compile error, a hyperlink is displayed on the error page that leads you to the corresponding file/line in IntelliJ. According to the documentation this can be done by adding 'play.editor="http://localhost:63342/api/file/?file=%s&line=%s"' to application.conf.
### Actual Behavior
In case of a compile error, no hyperlink is rendered on the error page.
### Reproducible Test Case
1. create a scala play 2.6.12 application
2. add 'play.editor="http://localhost:63342/api/file/?file=%s&line=%s"' to application.conf
3. produce a compile error 
4. witness that no hyperlink is displayed on the error page

**Comment by wsargent** [comment]
Hmm.

Does this work if you add https://chrome.google.com/webstore/detail/play-framework-tools/dchhggpgbommpcjpogaploblnpldbmen?hl=en-US and view the error?

**Comment by mkurz** [comment]
I think for `run` mode you need to set the option in `build.sbt`, like the docs suggest:
```
fork := true // required for "sbt run" to pick up javaOptions

javaOptions += "-Dplay.editor=http://localhost:63342/api/file/?file=%s&line=%s"
```

I also think setting the environment variable `PLAY_EDITOR` should work:
```
PLAY_EDITOR="http://localhost:63342/api/file/?file=%s&line=%s"
```

**Comment by nemoo** [comment]
@wsargent on one machine it works with the chrome plugin (on windows 10), on the other machine it doesn't work (on windows 7). That is why I was very happy to read in the play docs that it should work without any chrome plugin.

Just to be precise, this issue is really about the vanilla playframework implementation, not about the chrome plugin.

**Comment by nemoo** [comment]
@mkurz the docs say that alternatively you can also set play.editor from build.sbt:
```
fork := true // required for "sbt run" to pick up javaOptions

javaOptions += "-Dplay.editor=http://localhost:63342/api/file/?file=%s&line=%s"
```
I can confirm that this way of doing it works.

But this issue is about the other way that is described in the docs [here](https://www.playframework.com/documentation/2.6.x/IDE#Navigate-from-an-error-page-to-the-source-code)
I quote: 

> Enable the following line in application.conf to provide hyperlinks:
> play.editor="http://localhost:63342/api/file/?file=%s&line=%s"

**Comment by wsargent** [comment]
Right, that is https://github.com/playframework/playframework/pull/6372 which is a doc fix for https://github.com/playframework/playframework/pull/6009

So this was working at some point, but then it broke again.  It's probably the singleton DefaultHttpErrorHandler not being updated:

https://github.com/playframework/playframework/pull/6009/files#diff-b6bf57c572be293e773044c77bfc3581R22

**Comment by marcospereira** [comment]
@nemoo,

Some findings.

I could not reproduce the problem if I run `sbt run` from inside IntelliJ as describe a little above the docs you linked. See how the file is linked:

![screen shot 2018-04-05 at 3 02 19 pm](https://user-images.githubusercontent.com/4576/38383257-6bb93cb2-38e2-11e8-81c2-f3c523a50441.png)

Here is the `conf/application.conf`:

```
# This is the main configuration file for the application.
# https://www.playframework.com/documentation/latest/ConfigFile
play.editor="http://localhost:63342/api/file/?file=%s&line=%s"
```

But, when running using `sbt run` in a terminal, then I got the problem.

Not a solution, but it could help people looking at this problem.

**Comment by marcospereira** [comment]
> Why was this issue closed?

Reopening. It was closed by accident.

**Comment by mkurz** [comment]
I am pretty sure I found out what the real problem is here:

First of all we have to distinguish between
* failures produced from _compilation errors_ and
* failures produced by _exceptions_ thrown within the application.

When compilation errors occur, the app doesn't even start up, which means the `application.conf` won't even be loaded and any custom `play.editor` config will be ignored... That's why in such a case only the `javaOptions` and the nvironment variable (`PLAY_EDITOR`) approach work.

If a failure is caused by an exception (meaning compilation succeeded), the application most likely did start up already and `application.conf` was loaded and, of course, therefore a `play.editor` config set within the `application.conf` will be respected.

To avoid any confustion I suggest we just remove the part [of the docs](https://www.playframework.com/documentation/2.7.x/IDE#Navigate-from-an-error-page-to-the-source-code) which tells users to set the config within `application.conf` and they should just use the `javaOptions` or the environment variable (`PLAY_EDITOR`) approach, because they will always work, no matter if `application.conf` was loaded or not. We can also explicitly state the setting the config in `application.conf` won't work for compilation errors.
(BTW: I also tried `PlayKeys.devSettings += "play.editor" -> "http://localhost:63342/api/file/?file=%s&line=%s"` in `build.sbt`, however that will also be ignored when an compilation error occurs).

**Comment by MikeEngUSA** [comment]
I'm using v2.8.8, and it seems that it's no longer possible to fork the application. I have `fork := true` and define `javaOptions` in my `build.sbt` file, but, when executing `sbt run`, the web application runs in the same _JVM_ that _SBT_ is using, and none of the options I've defined take hold.

Has this capability been disabled? If so, is there a way to force the application, in dev mode, to be run in a forked _JVM_?

**Comment by mkurz** [comment]
>  ...it seems that it's no longer possible to fork the application ... when executing `sbt run`...

@MikeEngUSA Was that ever true in run mode? Can you say with which Play/sbt version it worked?  AFAIK forking `run` never worked in Play because they way it is implemented (overriding sbts `run` task if I remember correctly). Also see #7773
It works for `test` however.

**Comment by mkurz** [comment]
BTW: System properties set with `-D...` via `javaOptions` will be picked up even in run mode.
If you want to set anything else you can use the `.sbtopts` file like:
```
-J-Xmx2G
-J-Dfile.encoding=UTF-8
```
Also see `sbt -h`.
As alternative you can also use `.jvmopts` file:
```
-Xmx2G
-Dfile.encoding=UTF-8
```
